lib/src/indicator.dart

```diff
@@ -107,6 +125,23 @@ class CirclePainter extends CustomPainter {
     canvas.drawCircle(Offset(selectedX, radius), radius, _selectedPaint);
   }
 
+  void paintOval(Canvas canvas, Size size) {
+    var height = radius / 3;
+    var centerWidth = size.width / 2;
+    var startX = centerWidth - totalWidth / 2;
+    for (var i = 0; i < count ?? 0; i++) {
+      var x = startX + i * (radius * 2 + padding);
+      var rect = Rect.fromLTWH(x, 0, radius, height);
+      var rrect = RRect.fromRectAndRadius(rect, Radius.circular(height));
+      canvas.drawRRect(rrect, _circlePaint);
+    }
+
+    var selectedX = startX + page * (radius * 2 + padding);
+    var rect = Rect.fromLTWH(selectedX, 0, radius, height);
+    var rrect = RRect.fromRectAndRadius(rect, Radius.circular(height));
+    canvas.drawRRect(rrect, _selectedPaint);
```

Comment:
Use `drawOval` instead of `drawRRect` more conform to the meaning of shape.
If you need "RRect", you should use "RRect" instead of "Oval"  in `IndicatorShape`.

Reply:
using seperate class if RRectPainter for rendering RoundRectangle(earlier oval) on cnavas